### PR TITLE
Fix Block header part solidity library + test

### LINKED ISSUE #1328: Fix Block header part solidity library + test
Change fields in BlockHeaderPart to new version and write test for the library

### PR DESCRIPTION
**Pull request by Benzbeeb**
fixed #1328

### CODE REVIEW

**Review by taobun — DISMISSED** [review]

File: bridges/evm/contracts/BlockHeaderMerkleParts.sol [hunk 1]
```diff
@@ -30,8 +31,8 @@ library BlockHeaderMerkleParts {
     struct Data {
         bytes32 versionAndChainIdHash; // [1A]
         bytes32 timeHash; // [3]
-        bytes32 txCountAndLastBlockInfoHash; // [2B]
-        bytes32 consensusDataHash; // [2C]
+        bytes32 lastBlockIDAndOther; // [2B]
```
Comment (sorawit) [inline bridges/evm/contracts/BlockHeaderMerkleParts.sol:34]: Please update the comment to reflect the change

**Review by sorawit — CHANGES_REQUESTED** [review]

File: bridges/evm/contracts/BlockHeaderMerkleParts.sol [hunk 2]
```diff
@@ -45,31 +46,31 @@ library BlockHeaderMerkleParts {
         uint256 _blockHeight
     ) internal pure returns (bytes32) {
         return
-            Utils.merkleInnerHash( // [BlockHeader]
+            Utils.merkleInnerHash( // [Block Header]
```
Comment (sorawit) [inline bridges/evm/contracts/BlockHeaderMerkleParts.sol]: BlockHeader

File: bridges/evm/contracts/mocks/BlockHeaderMerklePartsMock.sol [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+
+contract BlockHeaderMerklePartsMock {
+    function getBlockHeader(
+        BlockHeaderMerkleParts.Data memory _self,
+        bytes32 _appHash,
+        uint256 _blockHeight
+    ) public pure returns (bytes32) {
+        return
+            BlockHeaderMerkleParts.getBlockHeader(
```
Comment (sorawit) [inline bridges/evm/contracts/mocks/BlockHeaderMerklePartsMock.sol:14]: move it up one line

**Review by sorawit — CHANGES_REQUESTED** [review]

File: bridges/evm/contracts/BlockHeaderMerkleParts.sol [hunk 4]
```diff
@@ -1,6 +1,7 @@
 pragma solidity 0.5.14;
 import {Utils} from "./Utils.sol";
 
+
 /// @dev Library for computing Tendermint's block header hash from app hash, time, and height.
 ///
 /// In Tendermint, a block header hash is the Merkle hash of a binary tree with 16 leaf nodes.
```
Comment (taobun) [inline bridges/evm/contracts/BlockHeaderMerkleParts.sol]: 14, [A] app_hash

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]

**Review by sorawit — APPROVED** [review]
looks good

**Pull request merged**